lib/msf/core/post/file.rb

```diff
@@ -322,6 +325,55 @@ def exist?(path)
 
   alias exists? exist?
 
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  #
+  def check_win_path_permissions(path, perms)
+    # Grab permissions of the path in question and extract relevant info using a regex.
+    # Also strip extra whitespace before comparison occurs.
+    permissions_file_raw = cmd_exec("icacls \"#{path}\"").to_s
```

Comment:
Since meterpreter and shell payloads can be run on older platforms that do not have `icals` we should probably check to see if it is available or guard the result here to detect if we got back `no command found`.

Reply:
Added code which should fix this in https://github.com/rapid7/metasploit-framework/pull/15987/commits/f3f907b3f630547b2d4aef4296545ee33e1a8fe1 though I'll still need to add in the `cacls` part of the code.